src/mprisplayer.cpp

```diff
@@ -70,6 +70,8 @@ MprisPlayer::MprisPlayer(QObject *parent)
 
 MprisPlayer::~MprisPlayer()
 {
+    // Unregister upon destruction
+    setServiceName(QStringLiteral(""));
```

Comment:
Hmm... Not sure this is the best way to unregister the service. setServiceName will call "registerService" which will print a warning if the service name is empty. The service will get unregistered, but I don't like the warning. Either split "unregisterService" as a separate function, or just copy-paste the couple of lines here.

(Also not nice to emit "serviceNameChanged" signal in the destructor.